ui/src/components/WebRTCVideo.tsx

```diff
@@ -144,27 +169,39 @@ export default function WebRTCVideo() {
     return () => {
       abortController.abort();
     };
-  }, [isPointerLockPossible, videoElm]);
+  }, [isPointerLockPossible]);
 
   const requestFullscreen = useCallback(async () => {
-    videoElm.current?.requestFullscreen({
+     if (!isFullscreenEnabled || !videoElm.current) return;
+
+    // per https://wicg.github.io/keyboard-lock/#system-key-press-handler
+    // If keyboard lock is activated after fullscreen is already in effect, then the user my 
+    // see multiple messages about how to exit fullscreen. For this reason, we recommend that
+    // developers call lock() before they enter fullscreen:
+    await requestKeyboardLock();
+    await requestPointerLock();
+
+    await videoElm.current.requestFullscreen({
       navigationUI: "show",
     });
+  }, [isFullscreenEnabled, requestKeyboardLock, requestPointerLock]);
 
-    // we do not care about pointer lock if it's for fullscreen
-    await requestPointerLock();
+  // setup to release the keyboard lock anytime the fullscreen ends
```

Comment:
This block ensures that when the browser fires the event in response to the full-screen mode changing, we will release the keyboard lock.